aten/src/ATen/cpu/vec/functional_base.h

```diff
@@ -27,6 +27,62 @@ inline scalar_t vec_reduce_all(
   return acc_arr[0];
 }
 
+template <typename scalar_t, typename Op>
+inline scalar_t vec_reduce_all(
+    const Op& vec_fun,
+    vec::Vectorized<scalar_t> acc_vec) {
+  using Vec = vec::Vectorized<scalar_t>;
+  return vec_reduce_all(vec_fun, acc_vec, Vec::size());
+}
+
+// SIMD version on horizontal reduce
+#if defined(CPU_CAPABILITY_AVX512) && !defined(_MSC_VER)
```

Comment:
ATen's vectorized struct `at::vec::Vectorized<>`  will be compiled 3 times: DEFAULT (the scalar version), AVX2(-mavx2 build) and AVX512 (-mavx512 build). The latter two are the real vectorized instructions and would eventually go to intrisincs. 

`_MSC_VER` refers to MSVC (windows c/c++ compiler), at the moment, `at::vec::Vectorized<>` only have the DEFAULT version on windows. Both AVX2 and AVX512 build are disabled on windows, as shown here:
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/cpu/vec/vec256/vec256_float.h#L18

GCC and Clang are OK, otherwise CIs would have failed.

Also this PR is only enabled with GCC version > 5:
```
#if defined(__GNUC__) && (__GNUC__ > 5) && !defined(_MSC_VER) && !defined(C10_MOBILE)
```
(gcc5 did not recognize the intrinsics `_mm256_cvtss_f32`, which causes CI failed previously)